coderd/workspaceagents.go

```diff
@@ -2030,78 +2030,25 @@ func (api *API) workspaceAgentsExternalAuth(rw http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	if listen {
```

Comment:
I hope my new flow is a bit easier to understand.

Basically there is 2 flows, `listen` and not listen. If `listen` is set to true, then the http handler will block until a valid auth token is found (this should probably be a websocket or something else idk).

Before `listen` and `not listen` did not share any code, which is dumb because on a success, their code paths should be identical. So this refactor makes the two cases share the same code execution path when a valid auth token is found.